### PR TITLE
core: fix Image prompt template hardcoded template format

### LINKED ISSUE #27411: Image prompt template does not respect template_format and assumes f-string
### Checked other resources

- [X] I added a very descriptive title to this issue.
- [X] I searched the LangChain documentation with the integrated search.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I am sure that this is a bug in LangChain rather than my code.
- [X] The bug is not resolved by updating to the latest stable version of LangChain (or the specific integration package).

### Example Code

```python
from langchain_core.prompts import ChatPromptTemplate, HumanMessagePromptTemplate

prompt = {
    'type': 'image_url',
    'image_url': {
        'url': 'data:{{ image_type }};base64,{{ image_data }}',
        'detail': 'low',
    },
}

data =  {
    'image_type': 'image/png',
    'image_data': 'dummyimageabcd12345'
}

template = ChatPromptTemplate.from_messages([('user', [prompt])], template_format='jinja2')
print(template.invoke(data))
```

### Error Message and Stack Trace (if applicable)

_No response_

### Description

The above code outputs:
```
messages=[HumanMessage(content=[{'type': 'image_url', 'image_url': {'url': 'data:{ image_type };base64, { image_data }', 'detail': 'low'}}])]
```
Rather than the expected output:
```
messages=[HumanMessage(content=[{'type': 'image_url', 'image_url': {'url': 'data:image/png;base64, dummyimageabcd12345', 'detail': 'low'}}])]
```
It looks like the template_format parameter is not being respected here ("f-string" is hard-coded): https://github.com/langchain-ai/langchain/blob/0ebddabf7ddafab30fefcbfae2148bf8e0fca42f/libs/core/langchain_core/prompts/chat.py#L556

Also, ImagePromptTemplate seems to be using string.format(): https://github.com/langchain-ai/langchain/blob/0ebddabf7ddafab30fefcbfae2148bf8e0fca42f/libs/core/langchain_core/prompts/image.py#L88

### System Info

System Information                                                   
------------------                                                   
> OS:  Linux                                                                 
> OS Version:  #17~22.04.1-Ubuntu SMP Tue Sep  3 16:11:52 UTC 2024           
> Python Version:  3.10.12 (main, Sep 11 2024, 15:47:36) [GCC 11.4.0]        
                                                                             
Package Information         
-------------------         
> langchain_core: 0.3.11    
> langchain: 0.3.3             
> langchain_community: 0.3.2      
> langsmith: 0.1.135              
> langchain_anthropic: 0.2.3      
> langchain_google_genai: 2.0.1   
> langchain_google_vertexai: 2.0.5                                           
> langchain_openai: 0.2.2                                                    
> langchain_text_splitters: 0.3.0                                            
                                                                             
Optional packages not installed
-------------------------------                                              
> langgraph                                                                  
> langserve                                                                  
                                                                             
Other Dependencies                                          
------------------                                          
> aiohttp: 3.10.10                                          
> anthropic: 0.32.0                                         
> anthropic[vertexai]: Installed. No version info available.
> async-timeout: 4.0.3           
> dataclasses-json: 0.6.7        
> defusedxml: 0.7.1              
> google-cloud-aiplatform: 1.70.0
> google-cloud-storage: 2.18.0
> google-generativeai: 0.8.3                                
> httpx: 0.27.2                                             
> httpx-sse: 0.4.0                                          
> jsonpatch: 1.33                                           
> langchain-mistralai: Installed. No version info available.
> numpy: 1.26.4  
> openai: 1.50.1 
> orjson: 3.10.7          
> packaging: 24.1         
> pillow: 10.3.0          
> pydantic: 2.9.2         
> pydantic-settings: 2.5.2
> PyYAML: 6.0.2           
> requests: 2.32.3        
> requests-toolbelt: 1.0.0 
> SQLAlchemy: 2.0.36                                                                                                                                      
> tenacity: 8.5.0                                                                                                                                         
> tiktoken: 0.7.0                                                                                                                                         
> typing-extensions: 4.12.2

### PR DESCRIPTION
**Pull request by chunkanglu**
Fixes #27411 

**Description:** Adds `template_format` to the `ImagePromptTemplate` class and updates passing in the `template_format` parameter from ChatPromptTemplate instead of the hardcoded "f-string".
Also updated docs and typing related to `template_format` to be more up-to-date and specific.

**Dependencies:** None

**Add tests and docs**: Added unit tests to validate fix. Needed to update `test_chat` snapshot due to adding new attribute `template_format` in `ImagePromptTemplate`.

### CODE REVIEW

**Review by vbarda — APPROVED** [review]
looks good -- thanks for your contribution!

File: libs/core/extended_testing_deps.txt [hunk 1]
```diff
@@ -1 +1,2 @@
 jinja2>=3,<4
+mustache>=0.1.4,<1
```
Comment (eyurtsev) [inline libs/core/extended_testing_deps.txt:2]: why was this added? we vendor the implementation of mustache

Comment (eyurtsev) [reply libs/core/extended_testing_deps.txt:2]: removing here https://github.com/langchain-ai/langchain/pull/27629

Comment (vbarda) [reply libs/core/extended_testing_deps.txt:2]: i added this because extended tests were failing for me https://github.com/langchain-ai/langchain/actions/runs/11448833821/job/31853195056
